testtools/files.go

```diff
@@ -129,3 +139,232 @@ func CheckFiles(t *testing.T, dir string, files []FileSpec) {
 		}
 	}
 }
+
+type TestGazelleGenerationArgs struct {
+	// name is the name of the test.
+	Name string
+	// testDataPath is the workspace relative path to the test data directory.
+	TestDataPath string
+	// gazelleBinaryDir is the workspace relative path to the location of the gazelle binary
+	// we want to test.
+	GazelleBinaryDir string
+	// gazelleBinaryName is the name of the gazelle binary target that we want to test.
+	GazelleBinaryName string
+
+	// The suffix for all test input build files. Includes the ".".
+	// Default: ".in", so input BUILD files should be named BUILD.in.
+	BuildInSuffix string
+
+	// The suffix for all test output build files. Includes the ".".
+	// Default: ".out", so out BUILD files should be named BUILD.out.
+	BuildOutSuffix string
+}
+
+func NewTestGazelleGenerationArgs() *TestGazelleGenerationArgs {
+	return &TestGazelleGenerationArgs{
+		BuildInSuffix:  ".in",
```

Comment:
Above it says that `BuildInSuffix` defaults to `.in`, if that's the case we shouldn't explicitly set the value here (same for `BuildOutSuffix`).